pennylane/circuit_graph.py

```diff
@@ -110,26 +110,31 @@ def __init__(self, ops, obs, wires, par_info=None, trainable_params=None):
         """dict[int, list[Operator]]: dictionary representing the quantum circuit as a grid.
         Here, the key is the wire number, and the value is a list containing the operators on that wire.
         """
+
+        self._indices = {}
+        # Store indices for the nodes of the DAG here
+
         self.wires = wires
         """Wires: wires that are addressed in the operations.
         Required to translate between wires and indices of the wires on the device."""
         self.num_wires = len(wires)
         """int: number of wires the circuit contains"""
         for k, op in enumerate(queue):
             meas_wires = wires or None  # cannot use empty wire list in MeasurementProcess
-            if isinstance(op, StateMP):
+            if isinstance(op, StateMP) and op.wires == Wires([]):
                 # State measurements contain no wires by default, but wires are
                 # required for the circuit drawer, so we recreate the state
                 # measurement with all wires
-                op = StateMP(wires=meas_wires)
-
+                copy_op = StateMP(wires=meas_wires)
             elif isinstance(op, SampleMP) and op.wires == Wires([]):
                 # Sampling without specifying wires is treated as sampling all wires
-                op = qml.sample(wires=meas_wires)
+                copy_op = qml.sample(wires=meas_wires)
+            else:
+                copy_op = op
 
             op.queue_idx = k  # store the queue index in the Operator
 
-            for w in op.wires:
+            for w in copy_op.wires:
```

Comment:
This is fine as is, but one thing I was doing was:  
```
for w in self.wires if len(op.wires) == 0 else op.wires:
```

No need for `copy_op`. less code. Makes is so you can remove the whole `if-elif-else` block.

That particularly important though.